### ISSUE TITLE
bicubic+border produces out-of-range values

### ISSUE THREAD

**Issue by beanduan22** [microsoft/onnxruntime#28090]
### Describe the issue

GridSample(bicubic, padding=border) clamps after neighbourhood lookup instead of per-sample

### To reproduce

```
import numpy as np
import onnx
from onnx import TensorProto, helper
import onnxruntime as ort

feat = np.arange(16, dtype=np.float32).reshape(1, 1, 4, 4)
grid = np.array([[[[-0.95, -0.95], [0.95, -0.95],
                   [-0.95,  0.95], [0.95,  0.95]]]], dtype=np.float32)

X = helper.make_tensor_value_info("X",    TensorProto.FLOAT, [1, 1, 4, 4])
G = helper.make_tensor_value_info("grid", TensorProto.FLOAT, [1, 1, 4, 2])
Y = helper.make_tensor_value_info("Y",    TensorProto.FLOAT, None)

node_bb = helper.make_node("GridSample", ["X","grid"], ["Y"],
    mode="bicubic", padding_mode="border", align_corners=0)
graph_bb = helper.make_graph([node_bb], "g", [X, G], [Y])
model_bb = helper.make_model(graph_bb, opset_imports=[helper.make_opsetid("", 16)])
sess_bb  = ort.InferenceSession(model_bb.SerializeToString(), providers=["CPUExecutionProvider"])
out_bicubic_border = sess_bb.run(None, {"X": feat, "grid": grid})[0]

node_bz = helper.make_node("GridSample", ["X","grid"], ["Y"],
    mode="bicubic", padding_mode="zeros", align_corners=0)
graph_bz = helper.make_graph([node_bz], "g", [X, G], [Y])
model_bz = helper.make_model(graph_bz, opset_imports=[helper.make_opsetid("", 16)])
sess_bz  = ort.InferenceSession(model_bz.SerializeToString(), providers=["CPUExecutionProvider"])
out_bicubic_zeros = sess_bz.run(None, {"X": feat, "grid": grid})[0]

print(f"bicubic+border output[0,0,0,:]: {out_bicubic_border[0,0,0,:]}")
print(f"bicubic+zeros  output[0,0,0,:]: {out_bicubic_zeros[0,0,0,:]}")
in_range = bool(np.all((out_bicubic_border >= -0.01) & (out_bicubic_border <= 15.01)))
print(f"Bicubic+border values in [0,15] range: {in_range}")
PASS = in_range
print(f"PASS={PASS}")
```

### Urgency

```
bicubic+border output[0,0,0,:]: [-0.5400001  2.6759987 12.323996  15.539993 ]
Bicubic+border values in [0,15] range: False
```

### Platform

Linux

### OS Version

Linux-6.17.0-20-generic-x86_64-with-glibc2.39

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.24.4

### ONNX Runtime API

Python

### Architecture

X86

### Execution Provider

Default CPU

**Comment by skottmckay** [comment]
This is expected behavior, not a bug. The `padding_mode="border"` attribute controls how **coordinates** outside the input grid are handled (by clamping them to the border), not the final interpolated value.

Bicubic interpolation uses a cubic kernel with negative lobes over a 4×4 pixel neighborhood. This inherently causes overshoot/ringing — interpolated values can exceed the range of the input data. This is a well-known property of cubic interpolation and is consistent with PyTorch's `torch.nn.functional.grid_sample(mode='bicubic', padding_mode='border')`, which produces the same out-of-range values.

If you need output values clamped to the input range, you can add a `Clip` node after the `GridSample` node:

```python
clip_node = helper.make_node("Clip", ["Y", "min", "max"], ["Y_clamped"])
```

Or if overshoot is unacceptable, use `mode="bilinear"` instead, which guarantees outputs within the convex hull of input values.
